src/Prefix.jl

```diff
@@ -173,7 +173,7 @@ function package(prefix::Prefix,
         @info("Tree hash of contents of $(basename(out_path)): $(tree_hash)")
     end
 
-    tarball_hash = archive_artifact(tree_hash, out_path; honor_overrides=false)
+    tarball_hash = _archive_artifact(tree_hash, out_path; honor_overrides=false)
```

Comment:
need to remove it here too

Reply:
From your [original comment](https://github.com/JuliaPackaging/BinaryBuilderBase.jl/pull/107#discussion_r566263696) I assumed we weren't using this keyword. What's the reason behind removing this? We'll probably want to switch back to the `Pkg.Artifacts` version at some point in the future

Reply:
I appreciate the explanation. Leaving the `honor_overrides` kwarg in place sounds like the ideal solution to me.